### PR TITLE
feat(condo): DOMA-9151 send hashed resident phones to marketing

### PR DESCRIPTION
**Pull request by nomerdvadcatpyat**
Sending hashed phone numbers of residents and contacts to the specified marketing email once a quarter. 
This is necessary for the marketing department to upload these numbers into marketing systems.

TODO:
- [x] fill MARKETING_EMAIL env variable

### CODE REVIEW

File: apps/condo/domains/resident/tasks/sendHashedResidentPhones.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+        stringifier.end()
+
+        taskLogger.info({ msg: 'Success load residents data', taskId })
+
+        const stream = fs.createReadStream(filename, { encoding: 'utf8' })
+        const status = await sendFileByEmail({ stream, filename, config: EMAIL_API_CONFIG, toEmail: MARKETING_EMAIL })
+
+        await redisClient.set(REDIS_KEY, dayjs().toISOString())
+        await rmfile(filename)
```
Comment (Alllex202) [inline apps/condo/domains/resident/tasks/sendHashedResidentPhones.js:72]: `sendFileByEmail` can throw error?
mb move `rmfile(filename)` to `finally`?

File: apps/condo/domains/resident/tasks/sendHashedResidentPhones.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+        taskLogger.info({ msg: 'File sent to email', taskId, status, toEmail: MARKETING_EMAIL })
+    } catch (error) {
+        taskLogger.error({ msg: 'Error in sendHashedResidentPhones', error, taskId })
+    }
+}
+
+// cron: At 00:00 on day-of-month 5 in every 3rd month
+const sendHashedResidentPhonesTask = createCronTask('sendHashedResidentPhones', '0 0 5 */3 *', sendHashedResidentPhones)
```
Comment (Alllex202) [inline apps/condo/domains/resident/tasks/sendHashedResidentPhones.js:81]: maybe choose another time? We already have several cron tasks for this time

File: apps/condo/domains/resident/tasks/helpers/sendHashedResidentPhones/getHashedResidentsAndContactsPhones.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+const { md5 } = require('@condo/domains/common/utils/crypto')
+const { RESIDENT } = require('@condo/domains/user/constants/common')
+
+
+const CHUNK_SIZE = 1000
+const PHONE_REGEXP = /^\+7\d{10}$/
+
+async function getHashedResidentsAndContactsPhones ({ lastSyncDate, toEmail, writePhoneCb }) {
+    const where = {
+        phone_contains_i: '+7',
```
Comment (Alllex202) [inline apps/condo/domains/resident/tasks/helpers/sendHashedResidentPhones/getHashedResidentsAndContactsPhones.js:17]:
```suggestion
        phone_starts_with_i: '+7',
```

File: apps/condo/domains/resident/tasks/helpers/sendHashedResidentPhones/getHashedResidentsAndContactsPhones.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    const lineWithIdentityHash = crypto.randomInt(maxLine)
+    let lineNum = 0
+
+    const chunkProcessor = async (chunk) => {
+        const phones = map(chunk, 'phone')
+
+        for (const phone of phones) {
+            if (PHONE_REGEXP.test(phone)) {
+                const formattedPhone = phone.slice(1)
+                writePhoneCb(formattedPhone)
```
Comment (Alllex202) [inline apps/condo/domains/resident/tasks/helpers/sendHashedResidentPhones/getHashedResidentsAndContactsPhones.js:46]: hashed phone?

File: apps/condo/domains/resident/tasks/helpers/sendHashedResidentPhones/getHashedResidentsAndContactsPhones.js [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+            }
+
+            if (lineNum === lineWithIdentityHash) {
+                // Hash to identify file: when and where was it sent. Example: 2024-05-30-example@example.com
+                const identityHash = md5(`${dayjs().format('YYYY-MM-DD')}-${toEmail}`)
+                writePhoneCb(identityHash)
+            }
```
Comment (Alllex202) [inline apps/condo/domains/resident/tasks/helpers/sendHashedResidentPhones/getHashedResidentsAndContactsPhones.js:53]: Is this for internal use?

File: apps/condo/domains/resident/tasks/helpers/sendHashedResidentPhones/getHashedResidentsAndContactsPhones.spec.js [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+const { getHashedResidentsAndContactsPhones } = require('./getHashedResidentsAndContactsPhones')
+
+
+describe('getHashedResidentsAndContactsPhones', () => {
+    const toEmail = 'test@example.com'
+
+    setFakeClientMode(index, { excludeApps: ['NextApp', 'AdminUIApp', 'OIDCMiddleware'] })
+
+
+    it('Correct data returns', async () => {
```
Comment (Alllex202) [inline apps/condo/domains/resident/tasks/helpers/sendHashedResidentPhones/getHashedResidentsAndContactsPhones.spec.js:19]: check that data is hashed

File: apps/condo/domains/resident/tasks/helpers/sendHashedResidentPhones/getHashedResidentsAndContactsPhones.js (hunk 5 above)
Comment (nomerdvadcatpyat) [reply apps/condo/domains/resident/tasks/helpers/sendHashedResidentPhones/getHashedResidentsAndContactsPhones.js:53]: Yes, it is for identify file if it was leaked

File: apps/condo/domains/resident/tasks/sendHashedResidentPhones.js
Comment (Alllex202) [inline apps/condo/domains/resident/tasks/sendHashedResidentPhones.js:1]: Should we provide the ability to use phone numbers for other countries (not +7)?

And what about the ability to disable this task? Maybe some third-party developers won't want to use it

Comment (nomerdvadcatpyat) [reply apps/condo/domains/resident/tasks/sendHashedResidentPhones.js:1]: > Should we provide the ability to use phone numbers for other countries (not +7)?

No

> And what about the ability to disable this task? Maybe some third-party developers won't want to use it

If they don't pass any of env variables (MARKETING_EMAIL), then task don't be executed

**Review by Alllex202 — APPROVED** [review]

**Pull request merged**